src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -138,8 +138,8 @@ def extract_strings(self) -> Tuple[Mapping[int, Set[str]], Set[str]]:
     @metrics.wraps("process_messages.reconstruct_messages")
     def reconstruct_messages(
         self,
-        mapping: Mapping[int, Mapping[str, int]],
-        bulk_record_meta: Mapping[str, Tuple[int, FetchType]],
+        mapping: Mapping[int, Mapping[str, Optional[int]]],
+        bulk_record_meta: Mapping[str, Metadata],
```

Comment:
Not sure I get why this is not broken by org. 
The same string can be in two orgs and we could be able to write it in one and not in the other.

Reply:
it's a long-standing bug. I pointed it out in sns channel a few months ago. we should fix it, thought somebody else was going to do it. will do now